### PR TITLE
Blog post added on "Controlling PF rules with SMF and IPS"

### PR DESCRIPTION
**Pull request by tmuehle**
Added my blog post on "Controlling PF rules with SMF and IPS" to the Blog_Supplements section.

The post is about using smf profiles and stencils to change pf rules by using includes in the pf.conf file.
All packed up in ips packages to be able to add and remove the rules esily.

Cheers
Thorsten

### CODE REVIEW

**Review by darrenmoffat — APPROVED** [review]
Very nice use of stencils and IPS to provide an easy way to deliver rule fragments for PF.

File: Blog_Supplements/Controlling_PF_Rules_with_SMF_and_IPS/example_files/firewall.stencil [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# We don't want PF to filter on loopback traffic by default.
+#
+# Filtering on loopback can interfere with zone installation and other
+# operations due to Solaris loopback optimizations. See the pf.conf(5)
+# manpage for guidance on how to enable it for your application.
+set skip on lo0
+
+## block everything unless told otherwise
+## and send TCP-RST/ICMP unreachable
+## for every packet which gets blocked
```
Comment (Sashan) [inline Blog_Supplements/Controlling_PF_Rules_with_SMF_and_IPS/example_files/firewall.stencil:36]: just a note `block return log` is fine if firewall is deployed in _friendly_ network. There is extra processing cost for each of those actions (`log` and `return`). If network is potentially hostile you may want to reconsider using just simple `block` instead. Silent discard punishes network scans.

File: Blog_Supplements/Controlling_PF_Rules_with_SMF_and_IPS/example_files/firewall.stencil [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+block return log
+
+## accept incoming icmp pkts e.g. ping
```
Comment (Sashan) [inline Blog_Supplements/Controlling_PF_Rules_with_SMF_and_IPS/example_files/firewall.stencil:39]: the rule accepts any ICMP message. I would go for ICMP echo only here.

**Review by Sashan — APPROVED** [review]
please consider my comments as nit-picks. I like what you've done.

**Comment by tmuehle** [comment]
@Sashan, no problem and thanks.

Just wanted to pick really simple examples. But you are right for sure.

**Review by joostpvh — COMMENTED** [review]
Looks great, there are a few typos in the README.md, but these are very minor.

**Pull request merged**